src/cpu/x64/matmul/brgemm_matmul_utils.cpp

```diff
@@ -740,7 +740,8 @@ struct matmul_avx512_blocking_params_t {
 
         bgmmc.use_buffer_c = is_buffer_c_required(
                 bgmmc.acc_dt, bgmmc.dst_dt, bgmmc.with_sum);
-        bgmmc.LDA = bgmmc.use_buffer_a || bgmmc.treat_transposed_A_as_plain
+        bgmmc.LDA = bgmmc.a_strides_changed || bgmmc.use_buffer_a
+                        || bgmmc.treat_transposed_A_as_plain
```

Comment:
Was the issue caught by benchdnn validation?

Reply:
Added a new testcase into `harness_matmul_regression_f32`.